ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/network/netty/DeFramer.java

```diff
@@ -168,12 +168,17 @@ protected void decode(final ChannelHandlerContext ctx, final ByteBuf in, final L
   }
 
   private Peer createPeer(final PeerInfo peerInfo, final ChannelHandlerContext ctx) {
-    InetSocketAddress remoteAddress = ((InetSocketAddress) ctx.channel().remoteAddress());
+    final InetSocketAddress remoteAddress = ((InetSocketAddress) ctx.channel().remoteAddress());
+    int port = peerInfo.getPort();
+    if (port == 0) {
+      // Most peers advertise a port of "0", just set a default best guess in this case
+      port = EnodeURL.DEFAULT_LISTENING_PORT;
```

Comment:
Instead of the default port perhaps we could use the remote port of the current communication?  We are already doing this with the remoteAddress.

Also, can these two values ever differ?  Such as if we have an eth loadbalancer or proxy in front of us (in case those ever exist for devp2p).

Reply:
I experimented with trying to use the remote port, but it doesn't match the listening port.  When the other nodes are initiating outbound connections, they're free to use whatever port they want for their side of the connection.

Reply:
Un-resolving this comment because I think its a good question and will be helpful to have it visible here as a kind of documentation.  Actually, I think I might add a comment in the codebase as well :D